### PR TITLE
fix group id and exclude subprojects from publishing and apidocs

### LINKED ISSUE #110: Fix maven coordinates and filter artifacts
Artifacts are being published to the Spring repo with the group ID `org.springframework.cloud.appbroker`. This should be just `org.springframework.cloud`. 

The modules `spring-cloud-app-broker-sample` and `spring-cloud-app-broker-acceptance-tests` are being published along with the library artifacts, but they should be excluded from publishing.

### PR DESCRIPTION
**Pull request by LittleBaiBai**
Resolves #110

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -286,6 +286,7 @@ configure(rootProject) {
 			project.sourceSets.main.compileClasspath
 		})
 
+		exclude '**/sample/**'
```
Comment (royclarkson) [inline build.gradle]: I wonder if we shouldn't narrow it down to the sample app package? Would this work?
```suggestion
		exclude 'org.springframework.cloud.appbroker.sample**'
```

Comment (LittleBaiBai) [reply build.gradle]: That would work. I was thinking this configuration would be applicable for other project with sample app as well. Is it common to have production code reside in a `sample` package?

Comment (royclarkson) [reply build.gradle]: We shouldn't have to worry about other production sample apps, since those shouldn't be published dependencies (artifacts).

**Comment by LittleBaiBai** [comment]
@royclarkson moved to pull request #136

File: build.gradle (hunk 1 above)
Comment (LittleBaiBai) [reply build.gradle]: @royclarkson applied your comment in pull request #136

**Pull request closed without merging**